validator/remote/src/integration-test/java/tech/pegasys/teku/validator/remote/typedef/handlers/ProduceBlockRequestTest.java

```diff
@@ -0,0 +1,276 @@
+
+    responseBodyBuffer.write(
+        spec.getGenesisSchemaDefinitions()
+            .getBlockContainerSchema()
+            .sszSerialize(blockContents)
+            .toArray());
+
+    mockWebServer.enqueue(
+        new MockResponse()
+            .setResponseCode(SC_OK)
+            .setHeader("Content-Type", MediaType.OCTET_STREAM)
+            .setBody(responseBodyBuffer));
+
+    final BLSSignature signature = blockContents.getBlock().getBody().getRandaoReveal();
+
+    final Optional<BlockContainer> maybeBlockContainer =
+        request.createUnsignedBlock(signature, Optional.empty());
+
+    assertThat(maybeBlockContainer).isPresent();
+
+    assertThat(maybeBlockContainer.get()).isEqualTo(blockResponse.getData());
+  }
+
+  @TestTemplate
+  public void shouldGetBlindedBlockContentsPostDenebAsJson() {
+    assumeThat(specMilestone).isEqualTo(DENEB);
+    final BlindedBlockContents blindedBlockContents =
+        dataStructureUtil.randomBlindedBlockContents(ONE);
+    ProduceBlockRequest.ProduceBlockResponse blockResponse =
+        new ProduceBlockRequest.ProduceBlockResponse(blindedBlockContents);
+
+    final String mockResponse = getExpectedBlockAsJson(specMilestone, true, true);
+
+    mockWebServer.enqueue(new MockResponse().setResponseCode(SC_OK).setBody(mockResponse));
+
+    final BLSSignature signature = blindedBlockContents.getBlock().getBody().getRandaoReveal();
+
+    final Optional<BlockContainer> maybeBlockContainer =
+        request.createUnsignedBlock(signature, Optional.empty());
+
+    assertThat(maybeBlockContainer).isPresent();
+
+    assertThat(maybeBlockContainer.get()).isEqualTo(blockResponse.getData());
+  }
+
+  @TestTemplate
+  public void shouldGetBlindedBlockContentsPostDenebAsSsz() {
+    assumeThat(specMilestone).isEqualTo(DENEB);
+    final BlindedBlockContents blindedBlockContents =
+        dataStructureUtil.randomBlindedBlockContents(ONE);
+    ProduceBlockRequest.ProduceBlockResponse blockResponse =
+        new ProduceBlockRequest.ProduceBlockResponse(blindedBlockContents);
+
+    responseBodyBuffer.write(
+        spec.getGenesisSchemaDefinitions()
+            .getBlindedBlockContainerSchema()
+            .sszSerialize(blindedBlockContents)
+            .toArray());
+
+    mockWebServer.enqueue(
+        new MockResponse()
+            .setResponseCode(SC_OK)
+            .setHeader(HEADER_EXECUTION_PAYLOAD_BLINDED, "true")
+            .setHeader("Content-Type", MediaType.OCTET_STREAM)
+            .setBody(responseBodyBuffer));
+
+    final BLSSignature signature = blindedBlockContents.getBlock().getBody().getRandaoReveal();
+
+    final Optional<BlockContainer> maybeBlockContainer =
+        request.createUnsignedBlock(signature, Optional.empty());
+
+    assertThat(maybeBlockContainer).isPresent();
+
+    assertThat(maybeBlockContainer.get()).isEqualTo(blockResponse.getData());
+  }
+
+  private String getExpectedBlockAsJson(
```

Comment:
maybe something like readExpectedResourceJson?

Reply:
Done, renamed it to `readExpectedJsonResource`